enzy_htp/structure/structure_region/residue_caps.py

```diff
@@ -0,0 +1,579 @@
+        """Create the default c-terminal version of the HCap with appropriate names."""
+        return [Atom(name='HP21', coord=[0.0, 0.0, 0.0], element='H' )]
+
+
+class CH3Cap(ResidueCap):
+    """A ResidueCap() which uses a -CH3 (methyl) as a cap. Adds a total of 4 atoms, with the 
+    
+    n-terminal version having atom names:       CP1, HP11, HP12, HP13
+    and c-terminal version having atom names:   CP2, HP21, HP22, HP23
+
+    Can generally be used for both n-terminal/c-teriminal and other non-amino acid bonds
+    """
+    FIXED_CHARGE_MAP:Dict[str, float] = {
+        # C-ter
+        "CP2" : -0.292800,
+        "HP21" : 0.097600,
+        "HP22" : 0.097600,
+        "HP23" : 0.097600,
+        # N-ter
+        "CP1" : -0.336900,
+        "HP11" : 0.112300,
+        "HP12" : 0.112300,
+        "HP13" : 0.112300,
+    } # TODO improve this for non ter.
+    """Fixed charge for a -CH3 attached to the N/C-ter
+    of a residue. Modified from NME and ACE in aminon/ct12.lib"""
+
+    CAP_BOND_DISTANCE:Dict[str, float] = {
+        "N" : 1.5,
+        "C" : 1.5,
+    }
+    """Maps the name of the link_atom to appropriate bond distance."""
+
+    def get_nterm_atoms(self) -> List[Atom]:
+        """Create the default n-terminal version of the CH3Cap with appropriate names."""
+        return [
+        Atom(name="CP1" , coord=[0.000,  0.000,  0.000], element='C'),
+        Atom(name="HP11", coord=[0.360, -1.029,  0.000], element='H'),
+        Atom(name="HP12", coord=[0.360,  0.514,  0.891], element='H'),
+        Atom(name="HP13", coord=[0.360,  0.514, -0.891], element='H'),]
+        
+    def get_cterm_atoms(self) -> List[Atom]:
+        """Create the default c-terminal version of the CH3Cap with appropriate names."""
+        return [
+        Atom(name="CP2" , coord=[0.000,  0.000,  0.000], element='C'),
+        Atom(name="HP21", coord=[0.360, -1.029,  0.000], element='H'),
+        Atom(name="HP22", coord=[0.360,  0.514,  0.891], element='H'),
+        Atom(name="HP23", coord=[0.360,  0.514, -0.891], element='H'),]
+
+    @property
+    def cap_type(self) -> str:
+        """What type of """
+        return "CH3"
+
+    @property
+    def net_charge(self) -> str:
+        """hard coded net charge"""
+        return 0
+
+    def anchor(self) -> None:
+        """Straightforward anchoring that is identical for both n-terminal and c-terminal capping."""
+        
+        p1 = np.array(self.link_atom.coord)
+        p2 = np.array(self.socket_atom.coord)
+
+        direction = p2 - p1
+        direction /= np.linalg.norm(direction)
+        rot_mat = rotation_matrix_from_vectors(np.array([1,0,0]), direction)
+        cap_bond_distance = self.CAP_BOND_DISTANCE[self.link_atom.name]
+
+        for aa in self.atoms:
+            pt = np.array(aa.coord)
+            pt = np.transpose(np.matmul(rot_mat, np.transpose( pt  )))
+            pt += (direction*cap_bond_distance) + p1 
+            aa.coord = pt
+
```

Comment:
Be great if we can put these math to math_hepler.py

Reply:
See above comment, but same idea.